### PR TITLE
pylint plugin to catch due/past-due deprecated calls

### PR DESCRIPTION
**Pull request by sivel**
##### SUMMARY
pylint plugin to catch due/past-due deprecated calls

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
test/sanity/pylint/plugins/deprecated.py

##### ANSIBLE VERSION

```
2.7
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: test/sanity/pylint/plugins/deprecated.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+
+ANSIBLE_VERSION = StrictVersion('.'.join(ansible_version_raw.split('.')[:3]))
+
+
+class AnsibleDeprecatedChecker(BaseChecker):
+    """Checks string formatting operations to ensure that the format string
+    is valid and the arguments match the format string.
+    """
```
Comment (mattclay) [inline test/sanity/pylint/plugins/deprecated.py]: This docstring needs to be updated.

**Review by mattclay — CHANGES_REQUESTED** [review]

**Comment by sivel** [comment]
```
lib/ansible/module_utils/aws/core.py:139:15: ansible-deprecated-no-version Display.deprecated call without a version
lib/ansible/module_utils/gcp.py:126:12: ansible-deprecated-version Deprecated version (2.5) found in call to Display.deprecated or AnsibleModule.deprecate
lib/ansible/module_utils/gcp.py:270:12: ansible-deprecated-version Deprecated version (2.5) found in call to Display.deprecated or AnsibleModule.deprecate
lib/ansible/plugins/callback/__init__.py:134:20: ansible-deprecated-no-version Display.deprecated call without a version
lib/ansible/plugins/loader.py:318:20: ansible-deprecated-no-version Display.deprecated call without a version
lib/ansible/plugins/strategy/__init__.py:829:16: ansible-deprecated-no-version Display.deprecated call without a version
```

**Comment by sivel** [comment]
We've got a clean pylint run: https://app.shippable.com/github/ansible/ansible/runs/82007/2/console

Going to add a disable for `ansible-deprecated-version` for normal execution, that we can remove after release in a PR to find out of date deprecated calls.

**Comment by sivel** [comment]
Removed `ansible-deprecated-version` to now look for 2.8 deprecations.

**Comment by sivel** [comment]
Ok, now that I have logged all deprecations for 2.8, I've added the ignore back in.

**Comment by sivel** [comment]
@mattclay @nitzmahone @abadger @ryansb 

Is there anything additional you would like to see done here?

Everything is passing right now (rebuilding right now due to stale_ci), and we will leave `ansible-deprecated-version` disabled by default, and enable via a PR after branching so that we can get a list of all of the deprecations that need to be removed for the next version.  (This likely will need added to the RM instructions)

**Comment by ryansb** [comment]
Can we make the flip automated by checking the base branch and flipping it on if there is `stable-*` in the branch? Perhaps checking the env var that's set by shippable for the base branch would be a good way to implement that.

**Comment by mattclay** [comment]
We need to keep tests passing on the devel branch, so we shouldn't change the version being checked automatically. Running the test needs to be done as part of the release process when we update the version number. The test should be updated, and the necessary fixes applied as a PR.

**Comment by sivel** [comment]
I've just added a script I used to create issues for the deprecated items found by this pylint plugin.

Run something like:

```
python create_deprecated_issues.py deprecated.txt
```

Where `deprecated.txt` is the output from pylint for the `ansible-deprecated-version` check.

**Comment by abadger** [comment]
@ryansb, the check needs to run on the devel branch after the previous version has its stable-X.Y branch created.  So the enhancement you propose would be testing something different.

**Review by mattclay — APPROVED** [review]

**Pull request merged**